### PR TITLE
fix(amplify_api): remove tabs from graphql document strings in androi…

### PR DESCRIPTION
**Pull request by ragingsquirrel3**
*Description of changes:*

Fixes https://github.com/aws-amplify/amplify-flutter/issues/1175 by removing tabs from graphql request document strings in android to prevent client errors that prevent the request from going out. Do not remove them when they are in quotes as that might have some unintended effects for users. Fix + some kotlin unit tests and an integration test.

I reformatted the kotlin files I touched, sorry for exploding the diff from that. Non-formatting changes noted in comments.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt [hunk 1]
```diff
@@ -30,16 +30,17 @@ object FlutterApiRequest {
...
     private const val HEADERS_KEY = "headers"
+    private val TABS_OUTSIDE_QUOTES_PATTERN = Regex("\t(?=(?:[^'\"`]*(['\"`])[^'\"`]*\\1)*[^'\"`]*\$)")
 
     // ====== Rest API ======
     fun getCancelToken(request: Map<String, Any>): String {
         try {
             return request[CANCEL_TOKEN_KEY] as String
         } catch (cause: Exception) {
             throw AmplifyException(
-                "The cancelToken request argument was not passed as a String",
```
Comment (ragingsquirrel3) [inline packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt:40]: Lots of formatting changes here, go straight to line 116 for interesting bit.

File: packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt [hunk 2]
```diff
@@ -42,15 +42,15 @@ import java.lang.reflect.Field
 import java.lang.reflect.Modifier
 
 const val underlyingMalformedException =
-    "AmplifyException{message=The graphQL document request argument " +
```
Comment (ragingsquirrel3) [inline packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt:45]: my non-formatting changes here start ~line 750.

File: packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt [hunk 3]
```diff
@@ -30,16 +30,17 @@ object FlutterApiRequest {
     private const val BODY_KEY = "body"
     private const val QUERY_PARAM_KEY = "queryParameters"
     private const val HEADERS_KEY = "headers"
+    private val TABS_OUTSIDE_QUOTES_PATTERN = Regex("\t(?=(?:[^'\"`]*(['\"`])[^'\"`]*\\1)*[^'\"`]*\$)")
```
Comment (dnys1) [inline packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt:33]: Can you add a comment with the unescaped version, or use a raw string?

\t(?=(?:[^'"`]*(['"`])[^'"`]*\1)*[^'"`]*$)

**Review by dnys1 — APPROVED** [review]

File: packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt (hunk 3 above)
Comment (dnys1) [reply packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt:33]: Also a small comment as to how it works?

Comment (ragingsquirrel3) [reply packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt:33]: Sure, will update.

File: packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt:40]: Can also just reference https://github.com/aws-amplify/amplify-flutter/pull/1178/files?diff=split&w=1

**Review by Jordan-Nelson — APPROVED** [review]

**Pull request merged**